File: sem 3/DS/sorting/bubble.cpp
#include <iostream>
#include <vector>

using namespace std;

int main() {
    vector<int> nums = {5, 3, 6, 7, 2, 9, 10, 14, 4};
    int end = nums.size() - 1;

    for (int end_ = end; end_ > 0; end_--) {
        for (int i=0; i < end_; i++) {
            if (nums[i] > nums[i+1]) {
                swap(nums[i], nums[i+1]);
            }
        }
    }

    for (int i=0; i<nums.size(); i++) {
        cout << nums[i] << " ";
    }
    cout << endl; 
    return 0;
}

File: sem 3/DS/sorting/bubble.py
nums = [5, 3, 6, 7, 2, 9, 10, 14, 4]
end = len(nums) - 1

for end_ in range(end, 0, -1):
    for i in range(0, end_):
        if nums[i] > nums[i+1]:
            nums[i], nums[i+1] = nums[i+1], nums[i]
            
print(nums)
    